File: Python/128_Longest_Consecutive_Sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        s=set(nums)
        maxlen=0
        for i in s:
            if i-1 not in s:
                l=1
                while i+1 in s:
                    l+=1
                    i+=1
                maxlen=max(l,maxlen)
        return maxlen

File: Java/128_Longest_Consecutive_Sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        HashSet<Integer> n=new HashSet<>();
        for(int i:nums){
            n.add(i);
        }
        int max=0;
        for(int i:nums){
           if(!n.contains(i-1)){
                int l=1;
                while(n.contains(i+l)){
                    l++;
                }
                max=Math.max(max,l);
           }
        }
        return max;

    }
}
